src/internal/git/pull.go

```diff
@@ -1,90 +1,87 @@
 package git
 
 import (
-	"context"
+	"errors"
+	"path/filepath"
 	"regexp"
 
 	"github.com/defenseunicorns/zarf/src/config"
 	"github.com/defenseunicorns/zarf/src/internal/message"
 	"github.com/defenseunicorns/zarf/src/internal/utils"
 	"github.com/go-git/go-git/v5"
 	"github.com/go-git/go-git/v5/plumbing"
-
-	"strings"
 )
 
 const onlineRemoteName = "online-upstream"
 
-func DownloadRepoToTemp(gitUrl string, spinner *message.Spinner) string {
+// DownloadRepoToTemp clones or updates a repo into a temp folder to perform ephemeral actions (i.e. process chart repos).
+func DownloadRepoToTemp(gitURL string, spinner *message.Spinner) string {
 	path, err := utils.MakeTempDir(config.CommonOptions.TempDirectory)
 	if err != nil {
 		message.Fatalf(err, "Unable to create tmpdir: %s", config.CommonOptions.TempDirectory)
 	}
 	// If downloading to temp, grab all tags since the repo isn't being
 	// packaged anyway, and it saves us from having to fetch the tags
 	// later if we need them
-	pull(gitUrl, path, spinner)
+	pull(gitURL, path, spinner, "")
 	return path
 }
 
-func Pull(gitUrl, targetFolder string, spinner *message.Spinner) (string, error) {
-	repoName, err := transformURLtoRepoName(gitUrl)
+// Pull clones or updates a git repository into the target folder.
+func Pull(gitURL, targetFolder string, spinner *message.Spinner) (string, error) {
+	repoName, err := transformURLtoRepoName(gitURL)
 	if err != nil {
-		message.Errorf(err, "unable to pull the git repo at %s", gitUrl)
+		message.Errorf(err, "unable to pull the git repo at %s", gitURL)
 		return "", err
 	}
 
 	path := targetFolder + "/" + repoName
-	pull(gitUrl, path, spinner)
+	pull(gitURL, path, spinner, repoName)
 	return path, nil
 }
 
-func pull(gitUrl, targetFolder string, spinner *message.Spinner) {
-	spinner.Updatef("Processing git repo %s", gitUrl)
-
-	gitCred := FindAuthForHost(gitUrl)
-
-	matches := strings.Split(gitUrl, "@")
-	onlyFetchRef := len(matches) == 2
-	cloneOptions := &git.CloneOptions{
-		URL:        matches[0],
-		Progress:   spinner,
-		RemoteName: onlineRemoteName,
-	}
+func pull(gitURL, targetFolder string, spinner *message.Spinner, repoName string) {
+	spinner.Updatef("Processing git repo %s", gitURL)
 
-	if onlyFetchRef {
-		cloneOptions.Tags = git.NoTags
+	gitCachePath := targetFolder
+	if repoName != "" {
+		gitCachePath = filepath.Join(config.GetCachePath(), filepath.Join(config.ZarfGitCacheDir, repoName))
 	}
 
-	// Gracefully handle no git creds on the system (like our CI/CD)
-	if gitCred.Auth.Username != "" {
-		cloneOptions.Auth = &gitCred.Auth
+	matches := gitURLRegex.FindStringSubmatch(gitURL)
+	if len(matches) == 0 {
+		// Unable to find a substring match for the regex
+		message.Fatalf("unable to get extract the repoName from the url %s", gitURL)
 	}
+	onlyFetchRef := matches[gitURLRegex.SubexpIndex("atRef")] != ""
+	gitURLNoRef := matches[gitURLRegex.SubexpIndex("proto")] + matches[gitURLRegex.SubexpIndex("hostPath")] + "/" +
```

Comment:
This line is kind of hard to read. We don't do a lot of it (and maybe we should), but go convention actually uses a lot of short variable names, which would help with readability here. Also you could use the function -> variable pattern used above here too or a variable function.  Something like 

```go
matches := gitURLRegex.FindStringSubmatch(gitURL)
idx := gitURLRegex.SubexpIndex

if len(matches) != 6 {
	// Unable to find a substring match for the regex
	message.Fatalf("unable to get extract the repoName from the url %s", gitURL)
}
onlyFetchRef := matches[idx("atRef")] != ""
gitURLNoRef := fmt.Sprintf("%s/%s%s", matches[idx("baseURL")], matches[idx("repo")], matches[idx("git")])
```

or

```go
matches := gitURLRegex.FindStringSubmatch(gitURL)
match := func(test string) string {
	return matches[gitURLRegex.SubexpIndex(test)]
}

if len(matches) != 6 {
	// Unable to find a substring match for the regex
	message.Fatalf("unable to get extract the repoName from the url %s", gitURL)
}
onlyFetchRef := match("atRef") != ""
gitURLNoRef := fmt.Sprintf("%s/%s%s", match("baseURL"), match("repo"), match("git"))
```

Reply:
I'll use the top one since I think it is a good compromise (I don't always like following weird helper functions around)

Reply:
haha yeah totally fair point..and agree, was just trying to throw ideas against the wall

Reply:
pushed up a change